ethportal-api/src/types/query_trace.rs

```diff
@@ -21,6 +21,8 @@ pub struct QueryTrace {
     pub origin: NodeId,
     /// Map of a node's ID to its corresponding `QueryResponse`
     pub responses: HashMap<NodeId, QueryResponse>,
+    /// Track if and how each node fails during a query
+    pub failures: HashMap<NodeId, QueryFailure>,
```

Comment:
@mrferris what do you think about this structure, for tracking failures types and times, for each peer?

(PR is unfinished, I just remember you saying you had opinions on this data structure, and figured I'd get early feedback)